core/src/test/java/gov/nist/secauto/metaschema/core/util/UriUtilsTest.java

```diff
@@ -0,0 +1,90 @@
+ * of warranty appears in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED 'AS IS' WITHOUT ANY WARRANTY OF ANY KIND, EITHER
+ * EXPRESSED, IMPLIED, OR STATUTORY, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY
+ * THAT THE SOFTWARE WILL CONFORM TO SPECIFICATIONS, ANY IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND FREEDOM FROM
+ * INFRINGEMENT, AND ANY WARRANTY THAT THE DOCUMENTATION WILL CONFORM TO THE
+ * SOFTWARE, OR ANY WARRANTY THAT THE SOFTWARE WILL BE ERROR FREE.  IN NO EVENT
+ * SHALL NIST BE LIABLE FOR ANY DAMAGES, INCLUDING, BUT NOT LIMITED TO, DIRECT,
+ * INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES, ARISING OUT OF, RESULTING FROM,
+ * OR IN ANY WAY CONNECTED WITH THIS SOFTWARE, WHETHER OR NOT BASED UPON WARRANTY,
+ * CONTRACT, TORT, OR OTHERWISE, WHETHER OR NOT INJURY WAS SUSTAINED BY PERSONS OR
+ * PROPERTY OR OTHERWISE, AND WHETHER OR NOT LOSS WAS SUSTAINED FROM, OR AROSE OUT
+ * OF THE RESULTS OF, OR USE OF, THE SOFTWARE OR SERVICES PROVIDED HEREUNDER.
+ */
+
+package gov.nist.secauto.metaschema.core.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+class UriUtilsTest {
+  private static final boolean VALID = true;
+  private static final boolean INVALID = false;
+
+  private static Stream<Arguments> provideValuesTestToUri() {
+    List<Arguments> values = new LinkedList<>() {
+      {
+        add(Arguments.of("http://example.org/valid", VALID));
+        add(Arguments.of("https://example.org/valid", VALID));
+        add(Arguments.of("http://example.org/valid", VALID));
+        add(Arguments.of("ftp://example.org/valid", VALID));
+        add(Arguments.of("ssh://example.org/valid", VALID));
+        add(Arguments.of("example.org/good", VALID));
+        add(Arguments.of("bad.txt", VALID));
+        add(Arguments.of("relative\\windows\\path\\resource.txt", VALID));
+        add(Arguments.of("C:\\absolute\\valid.txt", VALID));
+        add(Arguments.of("local/relative/path/is/invalid.txt", VALID));
+        add(Arguments.of("/absolute/local/path/is/invalid.txt", VALID));
+        add(Arguments.of("1;", VALID));
+        // These tests fail, they appear to be valid in Linux on GHA
+        // https://github.com/usnistgov/metaschema-java/actions/runs/7663271718/job/20885821561
+        // add(Arguments.of(":", INVALID));
+        // add(Arguments.of(">", INVALID));
+        // add(Arguments.of("<", INVALID));
+        // https://github.com/usnistgov/metaschema-java/actions/runs/7663616200/job/20886692509?pr=298
+        // add(Arguments.of(" ", INVALID));
+        // add(Arguments.of(" ?", INVALID));
```

Comment:
Now that I seem to know these negative tests won't work, I can remove them. I still cannot find a negative test with a string value unless I do some odd Unicode characters if you'd like.

```suggestion
```

Reply:
Removed in https://github.com/usnistgov/metaschema-java/pull/298/commits/7957c6dd0c7710ed95672369e18974847d086f3b.